### DISCUSSION TITLE
Is TasmoCompiler broken?

### DISCUSSION THREAD

**Discussion by NsinghP** [benzino77/tasmocompiler#352] [Q&A] [answered]
Hello, I am trying to launch online TasmoCompiler, however it is not working as it is used to. Now the browser does not give the pop up prompt. Neither does http://localhost:3000/ link work and at bottom right of browser it says "No Open Ports". Please see the attached image: 

![Screenshot (210)](https://user-images.githubusercontent.com/69470381/235354408-04182d14-0806-4b7b-8c33-c4ac0cca7399.png)

Above is screenshot from Windows11PC and Firefox. Is there a problem with Online Tasmocompiler or is this a problem on my local computer. Please help. 

Edit: Same thing happens when I try from a Linux Mint PC/Firefox:

![Screenshot from 2023-04-30 19-31-44](https://user-images.githubusercontent.com/69470381/235357123-c2833f1b-55f0-4f53-ba6a-77742ddadda8.png)

**Comment by benzino77** [comment, marked as answer]
It is a problem with gitpod. Give it a try in couple of hours.

**Reply by NsinghP** [reply]
> It is a problem with gitpod. Give it a try in couple of hours.

It has been over 24 hours, still the same problem persists. Is there a way to raise an issue with gitpod?

**Reply by benzino77** [reply]
I have no idea - you have to ask that question somewhere on gitpod side. It was working yesterday. Today it does not work again. What is interesting gitpod detects that there is open port but it is in state "detecting":
![image](https://user-images.githubusercontent.com/15658235/235454363-5acc4c1d-c67e-4f17-a203-43f14b7f3ce0.png)

You can always use other methods to run TasmoCompiler.

**Reply by NsinghP** [reply]
Yes, I can also see the same thing and it says detection for port 3000. I have contacted gitpod using their contact page form and they sent me an auto-reply email more than 8 hours ago but no response after that. 

I find other methods too complicated. TasmoCompiler is very easy to use. 


EDIT/Update: Someone has already logged an issue here: https://github.com/gitpod-io/gitpod/issues/17446 and
https://github.com/gitpod-io/gitpod/issues/17445

**Reply by NsinghP** [reply]
Here is the reply from Gitpod team:

Hi,

Thank you for reaching out. I'm sorry about the issue you're facing, we're still working on resolving it.

Meanwhile, you can use gp ports command in the terminal as a workaround. Please see the below link for more info:

    https://www.gitpod.io/docs/references/gitpod-cli#ports
    https://www.gitpod.io/docs/references/gitpod-cli#url

**Reply by benzino77** [reply]
So it should be enough to start new terminal on gitpod and run this one:
```sh
gp ports expose 3000
gp ports visibility 3000:public
```
and then get the url but executing:
```sh
$ gp ports list
| PORT |    STATUS     |                                URL                                 | NAME & DESCRIPTION |
|------|---------------|--------------------------------------------------------------------|--------------------|
| 3000 | open (public) | https://3000-benzino77-tasmocompiler-jq5zym1opmi.ws-eu96.gitpod.io |                    |
```
It may or may not work but you can give it a try.

**Reply by NsinghP** [reply]
> So it should be enough to start new terminal on gitpod and run this one:
> 
> ```shell
> gp ports expose 3000
> gp ports visibility 3000:public
> ```
> 
> and then get the url but executing:
> 
> ```shell
> $ gp ports list
> | PORT |    STATUS     |                                URL                                 | NAME & DESCRIPTION |
> |------|---------------|--------------------------------------------------------------------|--------------------|
> | 3000 | open (public) | https://3000-benzino77-tasmocompiler-jq5zym1opmi.ws-eu96.gitpod.io |                    |
> ```
> 
> It may or may not work but you can give it a try.

Thanks, but nothing happens after those commands.

**Reply by NsinghP** [reply]
This problem is now mentioned on GitPod Status page: 

![Screenshot (212)](https://user-images.githubusercontent.com/69470381/235573532-fb3cc3b6-a1c6-49ef-a244-db87fccb83c3.png)
https://www.gitpodstatus.com/

**Reply by NsinghP** [reply]
Current Problem is Fixed. Here is the update from Gitpod:

Identified - We have found that the new generation clusters do not include exposed ports information. We are rolling back and will be fixing it in the new clusters.
May 02, 2023 - 05:36 UTC
